main.go

```diff
@@ -579,7 +587,12 @@ func (c *controller) sync(ctx context.Context) {
 		}
 
 		c.replicas[hashring] = *sts.Spec.Replicas
-		statefulsets[hashring] = sts.DeepCopy()
+		if _, ok := statefulsets[hashring]; !ok {
+			// If not, initialize a new slice
+			statefulsets[hashring] = []*appsv1.StatefulSet{}
+		}
+		// Append the new value to the slice associated with the key
+		statefulsets[hashring] = append(statefulsets[hashring], sts.DeepCopy())
```

Comment:
I suggest adding logs to help debug issues. This is a lightweight and single pod service. Being a bit chatty should be fine. Some thing like 
```
level.Info(c.logger).Log("Hashring ", hashring, " got a statefulset: ", sts.Name)
```